c10/util/BFloat16-inl.h

```diff
@@ -7,15 +7,41 @@ namespace c10 {
 
 /// Constructors
 inline C10_HOST_DEVICE BFloat16::BFloat16(float value) {
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
+  auto bf = __float2bfloat16(value);
+  x = *reinterpret_cast<const unsigned short*>(&bf);
```

Comment:
To reinterpret memory this way, it's typically better to use a memcpy rather than a reinterpret cast on pointer; see https://en.cppreference.com/w/cpp/language/reinterpret_cast (we do compile with `-fno-strict-aliasing` so this will do the right thing, but it is better not to assume it.)

Reply:
I have updated this into 
```C++
#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000 && defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
  x = __bfloat16_as_ushort(__float2bfloat16(value));
#else
```
This uses intrinsic and will be available only at Ampere arch